src/contracts/atlas/ExecutionEnvironment.sol

```diff
@@ -121,179 +110,117 @@ contract ExecutionEnvironment is Base {
-                revert AtlasErrors.PostSolverFailed();
-            }
+            (success,) = _control().delegatecall(data);
 
-            success = abi.decode(data, (bool));
-            if (!success) {
-                revert AtlasErrors.IntentUnfulfilled();
-            }
+            if (!success) revert AtlasErrors.PostSolverFailed();
         }
 
-        uint256 endBalance = etherIsBidToken ? address(this).balance : ERC20(solverOp.bidToken).balanceOf(address(this));
-        uint256 netBid;
-
-        // Check if this is an on-chain, ex post bid search
-        if (_bidFind()) {
-            if (!config.invertsBidValue()) {
-                netBid = endBalance - startBalance; // intentionally underflow on fail
-                if (solverOp.bidAmount != 0 && netBid > solverOp.bidAmount) {
-                    netBid = solverOp.bidAmount;
-                    endBalance = etherIsBidToken ? netBid - solverOp.bidAmount : address(this).balance;
-                } else {
-                    endBalance = 0;
-                }
-            } else {
-                netBid = startBalance - endBalance; // intentionally underflow on fail
-                if (solverOp.bidAmount != 0 && netBid < solverOp.bidAmount) {
-                    netBid = solverOp.bidAmount;
-                    endBalance = etherIsBidToken ? solverOp.bidAmount - netBid : address(this).balance;
-                } else {
-                    endBalance = 0;
-                }
-            }
-        } else {
-            // Verify that the solver paid what they bid
-            if (!config.invertsBidValue()) {
-                // CASE: higher bids are desired by beneficiary (E.G. amount transferred in by solver)
-
-                // Use bidAmount arg instead of solverOp element to ensure that ex ante bid results
-                // aren't tampered with or otherwise altered the second time around.
-                if (endBalance < startBalance + bidAmount) {
-                    revert AtlasErrors.SolverBidUnpaid();
-                }
-
-                // Get ending eth balance
-                endBalance = etherIsBidToken ? endBalance - bidAmount : address(this).balance;
-            } else {
-                // CASE: lower bids are desired by beneficiary (E.G. amount transferred out to solver)
-
-                // Use bidAmount arg instead of solverOp element to ensure that ex ante bid results
-                // aren't tampered with or otherwise altered the second time around.
-                if (endBalance < startBalance - bidAmount) {
-                    // underflow -> revert = intended
-                    revert AtlasErrors.SolverBidUnpaid();
-                }
-
-                // Get ending eth balance
-                endBalance = etherIsBidToken ? endBalance : address(this).balance;
-            }
+        // bidValue is not inverted; Higher bids are better; solver must deposit >= bidAmount
+        if (!solverTracker.invertsBidValue) {
```

Comment:
Why do the floor and ceiling records get taken after the pre and post solver calls for !invertsBidValue, but the records get taken beforehand for invertsBidValue?

Reply:
This is a good question. So to rephrase, if mode is invertsBidValue, the change is measured over preSolverCall, solverCall, and postSolverCall. But if the mode is normal bid value, the change is just measured over solverCall. 

@thogard785 can you add some clarity around that difference?

Reply:
What you described would make sense, but that doesn't seem to be the implementation if i'm reading the code correctly. I worded it a bit weirdly at first. The below is the correct re-phrasing:

invertsBidValue the change is measured over preSolverCall and solverCall.
normal mode the change is measured over solverCall and postSolverCall.

Reply:
The idea is to have a hook that takes place outside of the balance check for each option so that we can transform or otherwise adjust the balances.  For the inverted bids, we wanted to be able to move inventory in so that we could then give it to the solver before the solver call, and for the normal bids we wanted to be able to move inventory out that we have received from the solver after the solver call.

Reply:
Cool makes sense, lets add comments to the code that document this.

Reply:
A consequence of this is that if a dapp calls contribute() in the postSolverHook in:
- normal mode this causes the contribution to be reduced from the bid amount calculation
- invertsBidBalue: this causes the contribution to be included in the bid amount calculation

Could this cause any issues?